lib/rubocop/cop/rails/duration_arithmetic.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks if Duration as added to or subtracted from Time.current.
+      #
+      # @example
+      #   # bad
+      #   Time.current - 1.minute
+      #   Time.current + 2.days
+      #
+      #   # fair
+      #   Date.yesterday + 3.days
+      #   created_at - 1.minute
+      #   3.days - 1.hour
+      #
+      #   # good
+      #   1.minute.ago
+      #   2.days.from_now
+      class DurationArithmetic < Base
+        extend AutoCorrector
+
+        MSG = 'Do not add or subtract duration.'
+
+        RESTRICT_ON_SEND = %i[+ -].freeze
+
+        DURATIONS = Set[:second, :seconds, :minute, :minutes, :hour, :hours,
+                        :day, :days, :week, :weeks, :fortnight, :fortnights]
+
+        # @!method duration_arithmetic_argument?(node)
+        #   Match duration subtraction or addition with current time.
+        #
+        #   @example source that matches
+        #     Time.current - 1.hour
+        #
+        #   @example source that matches
+        #     1.hour + ::Time.zone.now
+        #
+        #   @param node [RuboCop::AST::Node]
+        #   @yield operands and operator
+        def_node_matcher :duration_arithmetic_argument?, <<~PATTERN
+          {
+            (send $#time_current? ${ :+ :- } $#duration?)
+            (send $#duration? $:+ $#time_current?)
```

Comment:
This check is unnecessary, as `1.hour + Time.current` raises `TypeError: ActiveSupport::TimeWithZone can't be coerced into Integer`.